### DISCUSSION TITLE
requests to https://api.etsy.com/v3/public/oauth/token redirects inconsistently

### DISCUSSION THREAD

**Discussion by phillipdavidstearns** [etsy/open-api#1240] [Q&A]
Is anyone else experiencing redirects when requesting access tokens or refresh tokens from the https://api.etsy.com/v3/public/oauth/token endpoint?

The snippet of my python code implementing the access token request is as follows

```python
session = requests.Session()
session.headers={}
session.headers.update({
  "Accept": "application/json",
  "Content-Type": "application/x-www-form-urlencoded",
  "x-api-key": ETSY_API_KEY,
})

data={
  'grant_type': 'authorization_code',
  'client_id': ETSY_API_KEY,
  'redirect_uri': ETSY_CALLBACK_AUTH_URL,
  'code': CODE,
  'code_verifier': CODE_VERIFIER
}

response = session.post(
  url="https://api.etsy.com/v3/public/oauth/token",
  data=data,
  allow_redirects=False
)
```

Sometimes these requests return the access token on the first try. Other times the request will get redirected to an etsy sanctions policy page one or twice before succeeding.

If anyone can see why this might be the case or explain what this mysterious redirect behavior could be telling me, I'd appreciate the help!

**Comment by slider1234** [comment]
Try my Python code here instead https://github.com/etsy/open-api/discussions/1236
It will keep you from banging your head against the wall trying to figure it out. I automated the whole process.

**Reply by phillipdavidstearns** [reply]
Thanks, but the problems I was experiencing turned out to be an issue on Etsy's side. It was resolved shortly after raising it in a separate thread.
